src/client/eppo-client.ts

```diff
@@ -314,25 +317,13 @@ export default class EppoClient implements IEppoClient {
   private getSubjectVariationOverride(
     subjectKey: string,
     experimentConfig: IExperimentConfiguration,
-    valueType: ValueType,
-  ): EppoValue | null {
+    expectedValueType?: ValueType,
+  ): EppoValue {
     const subjectHash = md5(subjectKey);
-    const overridden =
+    const override = experimentConfig?.overrides && experimentConfig.overrides[subjectHash];
+    const typedOverride =
       experimentConfig?.typedOverrides && experimentConfig.typedOverrides[subjectHash];
-    if (overridden) {
-      switch (valueType) {
-        case ValueType.BoolType:
-          return EppoValue.Bool(overridden as unknown as boolean);
-        case ValueType.NumericType:
-          return EppoValue.Numeric(overridden as unknown as number);
```

Comment:
Not really sure why we needed multiple type assertions on this line: `overridden as unknown as number`

Reply:
🤨